apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx

```diff
@@ -0,0 +1,285 @@
+---
+title: Advanced command creation
+category: Slash commands
+---
+
+# Advanced command creation
+
+The examples we've covered so far have all been fairly simple commands, such as _`ping`_, _`server`_, and _`user`_ which all have standard static responses. However, there's much more you can do with the full suite of slash command tools!
+
+## Adding options
+
+Application commands can have additional _`options`_. Think of these options as arguments to a function, and as a way for the user to provide the additional information the command requires.
+
+<Alert title="Tip" type="success">
+	If you've already added options to your commands and need to know how to receive and parse them, refer to the [Parsing
+	options](./parsing-options) page in this section of the guide.
+</Alert>
+
+Options require at minimum a name and description. The same restrictions apply to option names as slash command names - 1-32 characters containing no capital letters, spaces, or symbols other than _`-`_ and _`_`_. You can specify them as shown in the _`echo`_ command below, which prompt the user to enter a String for the _`input`\_ option.
+
+<CH.Code>
+
+```js mark=6:12
+import { ApplicationCommandOptionType } from 'discord.js';
+
+export const data = {
+	name: 'echo',
+	description: 'Replies with your input!',
+	options: [
+		{
+			name: 'input',
+			type: ApplicationCommandOptionType.String,
+			description: 'The input to echo back',
+		},
+	],
+};
+```
+
+</CH.Code>
+
+## Option types
+
+By specifying the _`type`_ of an <DocsLink type="typedef" parent="ApplicationCommandOptionData" /> using the <DocsLink type="typedef" parent="ApplicationCommandOptionType" /> enum you are able to restrict what the user can provide as input, and for some options, leverage the automatic parsing of options into proper objects by Discord.
```

Comment:
`ApplicationCommandOptionType` is actually from discord-api-types.

Reply:
I agree, make DocsLink support it 😀

Reply:
Support added for this in 7b617bd22ecb4c009bdb0d807032134b51f6a855!